File: src/main/java/de/tectoast/emolga/selectmenus/ModalConfiguratorSelectMenu.java
package de.tectoast.emolga.selectmenus;

import de.tectoast.emolga.utils.automation.collection.ModalConfigurators;
import net.dv8tion.jda.api.events.interaction.component.SelectMenuInteractionEvent;

public class ModalConfiguratorSelectMenu extends MenuListener {
    public ModalConfiguratorSelectMenu() {
        super("modalconfigurator");
    }

    @Override
    public void process(SelectMenuInteractionEvent e, String name) {
        e.replyModal(
                ModalConfigurators.configurations.get(name).configurator().buildModal(Integer.parseInt(e.getValues().get(0)))
        ).queue();
    }
}

File: src/main/kotlin/de/tectoast/emolga/selectmenus/ModalConfiguratorSelectMenu.kt
package de.tectoast.emolga.selectmenus

import de.tectoast.emolga.utils.automation.collection.ModalConfigurators
import net.dv8tion.jda.api.events.interaction.component.StringSelectInteractionEvent

class ModalConfiguratorSelectMenu : MenuListener("modalconfigurator") {
    override fun process(e: StringSelectInteractionEvent, menuname: String?) {
        e.replyModal(
            ModalConfigurators.configurations[menuname]!!.configurator().buildModal(e.values[0].toInt())
        ).queue()
    }
}
